test/integration/controller/mpijob/mpijob_controller_test.go

```diff
@@ -302,6 +309,91 @@ var _ = ginkgo.Describe("Job controller for workloads when only jobs with queue
 		}, util.Timeout, util.Interval).Should(gomega.Succeed())
 	})
 
+	ginkgo.It("Should suspend a job if the parent workload does not exist", func() {
+		ginkgo.By("Creating the parent job which has a queue name")
+		parentJob := testingmpijob.MakeMPIJob(parentJobName, jobNamespace).
+			UID(parentJobName).
+			Queue("test").
+			Suspend(false).
+			Obj()
+		gomega.Expect(k8sClient.Create(ctx, parentJob)).Should(gomega.Succeed())
+
+		ginkgo.By("Creating the child job which uses the parent workload annotation")
+		childJob := testingjob.MakeJob(childJobName, jobNamespace).
+			OwnerReference(parentJobName, kubeflow.SchemeGroupVersionKind).
+			Suspend(false).
+			ParentWorkload("non-existing-parent-workload").
+			Obj()
+		gomega.Expect(k8sClient.Create(ctx, childJob)).Should(gomega.Succeed())
+
+		ginkgo.By("checking that the child job is suspended")
+		gomega.Eventually(func() *bool {
+			gomega.Expect(k8sClient.Get(ctx, childLookupKey, childJob)).Should(gomega.Succeed())
+			return childJob.Spec.Suspend
+		}, util.Timeout, util.Interval).Should(gomega.Equal(pointer.Bool(true)))
+	})
+
+	ginkgo.It("Should not suspend a child job if the parent job doesn't have a queue name", func() {
```

Comment:
It looks like you are mixing multiple cases here:
- doesn't have an owner
- unknown owner
- MPIJob without queue name

For the purpose of this integration test (in the mpi package) we should only have one test for the last case.
The rest can be in unit tests.

Reply:
done.